build-tools/automation/azure-pipelines.yaml

```diff
@@ -332,12 +332,12 @@ stages:
     - template: yaml-templates/fail-on-issue.yaml
 
 - stage: smoke_tests
-  displayName: Smoke Tests
+  displayName: macOS > Smoke Tests
```

Comment:
In order to reduce the verbosity in the GitHub checks that get reported back to the repo, maybe we should keep the stage names mostly unchanged and as simple as possible, while making job names more verbose.

The rename of the `Smoke Tests` stage to `macOS > Smoke Tests` is a bit odd, as the smoke test stage contains a job that runs on windows. Unless part of the intent behind the rename was to include stage dependencies in the name?

I would suggest something like this instead (leaving stage display names mostly unchanged):

Stage: `Smoke Tests`
Job: `macOS > Tests > APKs Classic`
Resulting GitHub status: `Xamarin.Android-PR (Smoke Tests macOS > Tests > Apks Classic`

We could also continue to use `macOS`, `Windows`, and `Linux` as stage names for the earlier build stages.

Reply:
I've reverted the `- stage > displayName` changes, so the historical stage names should be used once more.

I also quoted `displayName` values which contain `#`, as the "alt/hover text" for those entries were rendered as `… \# …`, which was _not_ the intent.

Reply:
Works for me, I think we may still end up with some "clunky" names but that's a fair trade off for better traceability.